src/main/java/com/neovisionaries/ws/client/WebSocketFrame.java

```diff
@@ -1199,15 +1198,17 @@ else if (frame.isContinuationFrame() == false)
 
         // Generate the first frame using the existing WebSocketFrame instance.
         // Note that the reserved bit 1 and the opcode are untouched.
-        byte[] payload = Arrays.copyOf(originalPayload, maxPayloadSize);
+        byte[] payload = new byte[maxPayloadSize];
+        System.arraycopy(originalPayload, 0, payload, 0, Math.min(originalPayload.length, maxPayloadSize));
         frame.setFin(false).setPayload(payload);
         frames.add(frame);
 
         for (int from = maxPayloadSize; from < originalPayload.length; from += maxPayloadSize)
         {
             // Prepare the payload of the next continuation frame.
             int to  = Math.min(from + maxPayloadSize, originalPayload.length);
-            payload = Arrays.copyOfRange(originalPayload, from, to);
```

Comment:
`Arrays.copyOfRange()` is not available in Java 1.5 that is targeted by this library.